### PR TITLE
[gprs] Fix pri_reset_context_properties crashes

### PR DESCRIPTION
**Pull request by tkenakka**
Fixes issues of ff63e9b0
There was e.g. crashes if current value (username, pwd etc) were empty but MBPI had new values.

### CODE REVIEW

File: ofono/src/gprs.c [hunk 1]
```diff
@@ -827,11 +827,14 @@ static gboolean pri_str_changed(const char *val, const char *newval)
 	return newval ? (strcmp(val, newval) != 0) : (val[0] != 0);
 }
 
-static gboolean pri_str_update(char *val, const char *newval)
+static gboolean pri_str_update(char *val, const char *newval,
+		const int maxlen)
 {
```
Comment (monich) [inline ofono/src/gprs.c]: The size is already checked by the time we get here

**Comment by monich** [comment]
LGTM except that the size check is unnecessary but that won't hurt I guess.

**Comment by tkenakka** [comment]
@monich Thanks. Yeah, decided to follow your contribution's line since it was there for some parameters. And like you say: won't hurt. Better safe than sorry.

File: ofono/src/gprs.c (hunk 1 above)
Comment (tkenakka) [inline ofono/src/gprs.c]: True, though not for "name".

Comment (monich) [inline ofono/src/gprs.c]: It wasn't invoked for "name" but I see that now it is.

**Comment by tkenakka** [comment]
Tagged

**Pull request merged**